phpBB/includes/acp/acp_permission_roles.php

```diff
@@ -48,7 +48,7 @@ function main($id, $mode)
 		$this->tpl_name = 'acp_permission_roles';
 
 		$submit = (isset($_POST['submit'])) ? true : false;
-		$role_id = $request->variable('role_id', 0);
+		$role_id = (int) $request->variable('role_id', 0);
```

Comment:
No need to type cast requests since them all pass via the `type_cast_helper` class.

In this specific case `, 0` defines INT, `, '0'` defines STRING instead.

You could see it using a simple test file to run from the root of your devs.

```
<?php
define('IN_PHPBB', true);
$phpbb_root_path = (defined('PHPBB_ROOT_PATH')) ? PHPBB_ROOT_PATH : './';
$phpEx = substr(strrchr(__FILE__, '.'), 1);
include($phpbb_root_path . 'common.' . $phpEx);

global $request;

$test = $request->variable('blah', 0);

var_dump($test);

```